core/shared/src/main/scala/monocle/Lens.scala

```diff
@@ -251,6 +251,10 @@ object Lens {
   def codiagonal[S]: Lens[S \/ S, S] =
     PLens.codiagonal
 
+  /** creates a [[Lens]] from a Store comonad coalgebra */
+  def storeFCoalg[F[_], S, A](f: S => F[S])(implicit CS: ComonadStore[F, A]): Lens[S, A] =
```

Comment:
I like `ComonadStore`, since it decouples particular representations, but given the lack of instances for it, I find your suggestion more than reasonable. Thereby, I've replaced this constructor by an `IndexedStore` version and moved it to `PLens`.